README.md

```diff
@@ -1,21 +1,97 @@
-### Assigning students to courses
+# Application - Assigning students to courses
 
-This program is a simple app that stores courses and students data and assigns students to a course.
+This program is a simple app that stores courses and students data and can assigns students to a course. It is a simple REST API written in python using Flask.
 
-IT is a simple REST API written in python using Flask.
+## REST API
 
-## What is REST API and Flask
+A REST (REpresentational State Transfer) API (Application Programming Interface) is a transfer of resources (data) from a Server to a Client over the internet. Through the REST API, the client can perform a GET, POST, PUT, DELETE request to the server.
 
-> what api provides to client
+A client uses an API to request for resources from the server. A server has the resources and transfers it to the client through an API. A resource can be any object for example, user, photo, number. An API allows one software to talk to another. It is a procedure allowing a client to access the data from a server. 
 
+An example of an API is "https://github.com/StephenDsouza90/demo-crud-sqlite-python-app" where:
+ 
+github.com/StephenDsouza90 - Endpoint URL
 
-##  The set up
+demo-crud-sqlite-python-app - Parameter
 
+## Flask
+
+Flask is a mirco web framework written in Python. 
+
+>>>>>> Need to write more on flask
+
+## Set up of rest.py 
+
+This app runs on a localhost (my computer) with an IP address of '0.0.0.0' (which maps to my computer) and a port 8080 as an endpoint.
+
+Example: 
+
+github.com - Localhost
+
+StephenDsouza90 - Endpoint
 
 ## How the app works
-> details on every function
-> its routing
-> how url path maps to the function
 
+In this app, the courses and students data has already been created in the courses_data and students_data objects respectively. The assignment_data has an empty list through which the client can alter it as required.
+
+### create_app()
+
+The create_app function has the following methods for
```

Comment:
So the `create_app()` function is also responsible for defining the routes and these routes would be mapped to the functions and these functions would then be responsible to process client requests and send back responses.

Reply:
change made